### ISSUE TITLE
ipfs swarm connect --single

### ISSUE THREAD

**Issue by whyrusleeping** [ipfs/go-ipfs#4191]
We need a way to tell `ipfs swarm connect` that it should *only* use the address we give to it. Too many times I try to connect to a peer with `ipfs swarm connect /definitely/a/valid/address` and get dial backoffed because ipfs tried 100 other addresses that we're the one i gave it. This is also important in testing other things, I want to test out relay stuff, but often times i try to connect to a peer via relay and get a normal tcp direct connection.

**Comment by Stebalien** [comment]
I'd expect `ipfs swarm connect` to *only* use the address we give it. To use any known address and/or the given address, I'd call `ipfs swarm connect /ip4/.../tcp/.../ipfs/QmId /ipfs/QmId ...` (that is, specify that we're willing to connect to the peer via any means necessary by passing an address without any route information).

**Comment by whyrusleeping** [comment]
Yeah, I think its fair to say that the current way the code works is buggy, and needs to be fixed.

**Comment by whyrusleeping** [comment]
potential duplicate of issue here: https://github.com/ipfs/go-ipfs/issues/3658

**Comment by kjzz** [comment]
hey @whyrusleeping , Has this issue finished?If not , i am willing to do it.It maybe will help me learn more about swarm.

**Comment by Stebalien** [comment]
This has not been fixed. The current thinking is that we'll add some option to the *context* saying "only dial these addresses when connecting to this peer".

**Comment by kjzz** [comment]
Thxs for replying. You mean that I shold add a option such as `--allow-others`(the option name must not the best one,anther will be better) . This option can determine whether to connect to other nodes when use command `ipfs swarm connect`.WDYT?

**Comment by Stebalien** [comment]
SGTM. The tricky part will be wiring this all the way down to the swarm/network.

As for naming, maybe `--routing=false` or something like that? Really, that's what we're doing. We're turning peer routing off. However, that may not be explicit enough so I'm open to suggestions :smile:.

**Comment by kjzz** [comment]
hey @Stebalien ,  should i clear all peer in peerstore when i use command `--routing=false`?

**Comment by Stebalien** [comment]
Ideally not, that's the tricky part. My current thinking is that we'd add options to the context. We'd probably want one to disable routing and one to explicitly specify which addresses we want to dial. However, I don't have any concrete ideas of how to implement that.
